src/alfasim_sdk/result_reader/aggregator.py

```diff
@@ -149,28 +157,38 @@ class BaseUQMetaData:
         The unit of a dimensionless property (-, %)
     """
 
-    property_id: str = attr.ib(validator=attr.validators.instance_of(str))
-    trend_id: str = attr.ib(validator=attr.validators.instance_of(str))
-    category: str = attr.ib(validator=attr.validators.instance_of(str))
-    network_element_name: Optional[str] = attr.ib(
-        validator=attr.validators.optional(attr.validators.instance_of(str))
-    )
-    position: Optional[float] = attr.ib(
-        validator=attr.validators.optional(attr.validators.instance_of(float))
-    )
-    position_unit: Optional[str] = attr.ib(
-        validator=attr.validators.optional(attr.validators.instance_of(str))
-    )
-    unit: str = attr.ib(validator=attr.validators.instance_of(str))
+    property_id: str
+    trend_id: str
+    category: str
+    network_element_name: Optional[str]
+    position: Optional[float]
+    position_unit: Optional[str]
+    unit: str
 
 
-@attr.s(slots=True, hash=False)
+@dataclasses.dataclass(slots=True, frozen=True)
+class UPOutputKey:
+    property_name: str
+    element_name: str
+
+    def __str__(self):
+        return f"{self.property_name}@{self.element_name}"
+
+    @classmethod
+    def from_string(cls, raw_output_key: str) -> Self:
+        r = re.match(r"^(.*)@(.*)", raw_output_key)
+        assert r is not None, f"Invalid output key: {raw_output_key}"
+        property_name, element_name = r.groups()
+        return cls(property_name, element_name)
+
+
+@dataclasses.dataclass(slots=True, frozen=True)
```

Comment:
This used `hash=False` previously... not sure why though, making it frozen seems reasonable.

Reply:
Yeah I'm also not sure why, but at least for our internal use cases frozen works fine so I guess we're good.